optimize.sh

```diff
@@ -4,20 +4,45 @@ command -v shellcheck > /dev/null && shellcheck "$0"
 
 export PATH=$PATH:/root/.cargo/bin
 
-# This parameter allows us to mount a folder into docker container's "/code"
-# and build "/code/contracts/mycontract".
-contractdir="$1"
-echo "Building contract in $(realpath -m "$contractdir")"
+mkdir -p artifacts
+contractdir=${1:-.}
 
-(
-  cd "$contractdir"
 
-  # Linker flag "-s" for stripping (https://github.com/rust-lang/cargo/issues/3483#issuecomment-431209957)
-  # Note that shortcuts from .cargo/config are not available in source code packages from crates.io
+# There are two cases here
+# 1. All contracts (or one) are included in the root workspace  (eg. `cosmwasm-template`, `cosmwasm-examples`, `cosmwasm-plus`)
+#    In this case, we pass no argument, just mount the proper directory.
+# 2. Contacts are excluded from the root workspace, but import relative paths from other packages (only `cosmwasm`).
+#    In this case, we mount root workspace and pass in a path `docker run <repo> ./contracts/hackatom`
+if [ "$contractdir" = "." ]; then
+  echo "Building all contracts in repo"
   RUSTFLAGS='-C link-arg=-s' cargo build --release --target wasm32-unknown-unknown --locked
-  wasm-opt -Os target/wasm32-unknown-unknown/release/*.wasm -o contract.wasm
+else
+  # This parameter allows us to mount a folder into docker container's "/code"
+  # and build "/code/contracts/mycontract".
+  echo "Building contract in $(realpath -m "$contractdir")"
+  (
+    cd "$contractdir"
+
+    # Linker flag "-s" for stripping (https://github.com/rust-lang/cargo/issues/3483#issuecomment-431209957)
+    # Note that shortcuts from .cargo/config are not available in source code packages from crates.io
+    RUSTFLAGS='-C link-arg=-s' cargo build --release --target wasm32-unknown-unknown --locked
+  )
+fi
```

Comment:
I think we don't need the `if` and duplication of the build command since `cd .` works in the first case as well. A text like `echo "Building all contracts in $(realpath -m "$contractdir")"` works for all cases.

Reply:
I will revisit. I needed it earlier, then kept refactoring the code to minimize the if. Maybe I don't need it at all... Good eye